modules/data_release/php/data_release.class.inc

```diff
@@ -51,70 +54,83 @@ class Data_Release extends \NDB_Menu_Filter
-        $data_release_ids      = array();
-        $data_release_versions = array();
+        // Get list of users in the database
+        $list_of_users = $this->getUsersList();
 
-        foreach ($versionedFiles as $versionName=>$versionFiles) {
-            foreach ($versionFiles as $fileID=>$fileName) {
-                $data_release_ids[$fileID] = $fileName . " - " . $versionName;
-            }
-            $data_release_versions[$versionName] = $versionName;
+        // Get list of data release versions and filenames
+        $vFiles = $this->getVersionedFiles();
 
+        $filenames_options = array();
+        $versions_options  = array();
+
+        foreach ($vFiles as $versionName => $filesArray) {
+            foreach ($filesArray as $fileID => $fileName) {
+                $filenames_options[$fileID] = "$fileName - $versionName";
+            }
+            $versions_options[$versionName] = $versionName;
         }
 
-        $manage_permissions = $this->getUserVersionPermissions($versionedFiles, $DB);
+        // Get list of permissions per data release versions
+        $manage_permissions = $this->getUserVersionPermissions($vFiles);
+
+        $this->fieldOptions = [
+                               'users'              => $list_of_users,
+                               'versions'           => $versions_options,
+                               'filenames'          => $filenames_options,
+                               'manage_permissions' => $manage_permissions,
+                              ];
+    }
+
+
+    /**
+     * Greps the list of users available in the users database table.
+     *
+     * @return array Array of user ID and user name
+     */
+    function getUsersList()
+    {
+        $Factory = \NDB_Factory::singleton();
+        $DB      = $Factory->Database();
+
+        $userids = $DB->pselectColWithIndexKey(
+            "SELECT ID, UserID FROM users",
+            array(),
+            "ID"
+        );
 
-        $this->tpl_data['user_upload']      = $userUpload;
-        $this->tpl_data['user_edit_access'] = $userEditAccess;
-        $this->tpl_data['userids']          = $userids;
-        $this->tpl_data['data_release_ids'] = $data_release_ids;
-        $this->tpl_data['data_release_versions'] = $data_release_versions;
-        $this->tpl_data['manage_permissions']    = $manage_permissions;
+        return $userids;
     }
 
+
     /**
      * Gets the array of release versions with their corresponding files.
      *
      * This function is used for the initial load as well as for comparison upon
      * update to ensure no values are added or deleted unexpectedly.
      *
-     * @param \Database $DB Database to use for querying
-     *
      * @return array Associative array of $versions=>(files associate with version)
      */
-    function getVersionedFiles(\Database $DB)
+    function getVersionedFiles()
     {
+        $Factory = \NDB_Factory::singleton();
+        $DB      = $Factory->Database();
```

Comment:
This looks like a step backwards

Reply:
oh noooo. I prefer moving forward too. Changed it to \Database::singleton() (guessing this is the forward move?)

Reply:
I think Dave meant that this function should take a Database as a parameter rather than using the factory or singleton.

Reply:
I'm not sure why this is better though, haha